src/enforester.js

```diff
@@ -1326,25 +1328,20 @@ export class Enforester {
 
   enforestNewExpression() {
     this.matchKeyword('new');
-    let callee;
     let lookahead = this.peek();
-    if (this.isKeyword(lookahead, 'new')) {
-      callee = this.enforestNewExpression();
-    } else if (this.isKeyword(lookahead, 'super')) {
-      callee = this.enforestLeftHandSideExpression({ allowCall: false });
-    } else if (this.isParens(lookahead) || this.isBrackets(lookahead) || this.isBraces(lookahead) || this.isTemplate(lookahead)) {
-      callee = this.enforestExpressionLoop();
-    } else if (this.isPunctuator(lookahead, '.') && this.isIdentifier(this.peek(1), 'target')) {
+    if (this.isPunctuator(lookahead, '.') && this.isIdentifier(this.peek(1), 'target')) {
       this.advance();
       this.advance();
       return new Term('NewTargetExpression', {});
+    } else if(this.isIdentifier(lookahead)){
+      this.term = new Term('IdentifierExpression', { name : this.enforestIdentifier() });
     } else {
-      callee = new Term('IdentifierExpression', { name : this.enforestIdentifier() });
+      this.term = this.enforestLeftHandSideExpression({ allowCall: false });
     }
 
+    // (new a``) is a NewExpression while (new a()``) is a TemplateExpression
     if(this.isTemplate(this.peek())) {
-      this.term = callee;
-      callee = this.enforestLeftHandSideExpression({ allowCall: false });
+      this.term = this.enforestLeftHandSideExpression({ allowCall: false });
     }
```

Comment:
I'm not following why you need separate cases for identifier expressions and templates. Aren't these both handled by `enforestLeftHandSideExpression`?

btw, the main reason I suggested moving the cases to `enforestLHSE` is that I think we actually need to use `enforestLHSE` in a few other places (e.g. class extends clause) so it should be as general as we need. We can fix up those other cases once this PR lands; we just need the fully general `enforestLHSE`.